test/Transactions/Orleans.Transactions.Azure.Test/AzureTransactionalStateStorageTests.cs

```diff
@@ -19,17 +19,23 @@
 
 namespace Orleans.Transactions.Azure.Tests
 {
-    public class TestState : TestKit.ITestState
+    public class TestState : IEquatable<TestState>
     {
-        public int state { get; set; } = 1;
+        public static Random Seed = new Random();
+        public int State { get; set; }
+
+        public bool Equals(TestState other)
+        {
+            return this.State.Equals(other.State);
+        }
     }
 
     public class AzureTransactionalStateStorageTests : TransactionalStateStorageTestRunnerxUnit<TestState>, IClassFixture<TestFixture>
     {
         private const string tableName = "StateStorageTests";
         private const string partition = "testpartition";
         public AzureTransactionalStateStorageTests(TestFixture fixture, ITestOutputHelper testOutput)
-            :base(()=>StateStorageFactory(fixture), ()=>new TestState(), fixture.GrainFactory, testOutput)
+            :base(()=>StateStorageFactory(fixture), ()=>new TestState(){State = TestState.Seed.Next()}, fixture.GrainFactory, testOutput)
```

Comment:
Should state factory have some hint for making different states deterministically, rather than relying on randomness?  Like maybe take in optional integer?  As is, either all expected states are the same as each other and the default state, allowing for a wide range of bugs to go undetected, or the factory must introduce randomness, which may make test failures intermittent and hard to reproduce.

Reply:
I agree with this.  My implementation used a int seed parameter to make the output deterministic.

---
In reply to: [273162605](https://github.com/dotnet/orleans/pull/5514#discussion_r273162605) [](ancestors = 273162605)

Reply:
will do